stan-ballerina/tests/nats_streaming_consumer_tests.bal

```diff
@@ -0,0 +1,387 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.'string;
+import ballerina/lang.runtime as runtime;
+import ballerina/log;
+import ballerina/test;
+
+const CONSUMER_SERVICE_SUBJECT_NAME = "nats-streaming-consumer-service";
+const ACK_SUBJECT_NAME = "nats-streaming-ack";
+const DUMMY_SUBJECT_NAME = "nats-streaming-dummy";
+const INVALID_SUBJECT_NAME = "nats-streaming-invalid";
+const SERVICE_NO_CONFIG_NAME = "nats-streaming-service-no-config";
+const QUEUE_SUBJECT_NAME = "nats-streaming-queue";
+const DURABLE_SUBJECT_NAME = "nats-streaming-queue";
+
+string receivedConsumerMessage = "";
+string receivedAckMessage = "";
+string noConfigServiceReceivedMessage = "";
+string receivedQueueMessage = "";
+string receivedDurableMessage = "";
+
+boolean ackNegativeFlag = false;
+boolean invalidServiceFlag = true;
+
+@test:Config {
+    dependsOn: [testProducer],
+    groups: ["nats-streaming"]
+}
+public function testConsumerService() {
+    string message = "Testing Consumer Service";
+    Listener sub = checkpanic new(DEFAULT_URL);
+    Client newClient = checkpanic new(DEFAULT_URL);
+    checkpanic sub.attach(consumerService);
```

Comment:
Better if you could replace `checkpanic` with `check` and add an `error?` return statement to the test.